addons/hr_recruitment_skills/models/hr_job.py

```diff
@@ -0,0 +1,22 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, _
+
+class HrJob(models.Model):
+    _inherit = "hr.job"
+
+    skill_ids = fields.Many2many('hr.skill', string="Expected Skills")
+
+    def action_search_matching_applicant(self):
+        action = self.env['ir.actions.actions']._for_xml_id('hr_recruitment.crm_case_categ0_act_job')
+        action['name'] = _("Matching Applicants")
+        action['views'] = [(self.env.ref('hr_recruitment_skills.crm_case_tree_view_inherit_hr_recruitment_skills').id, 'tree'), (False, 'form')]
+        action['context'] = {'active_id': self.id}
+        action['domain'] = [
```

Comment:
so, as for Geoffrey's response, we should return not active applicant's as well.. and also when we redirect to the view, add grouping by stage